### PR TITLE
Fix cancellation for GraphQL requests

### PR DESCRIPTION
**Pull request by eseliger**
Due to a bug in the fromFetch method, cancellation is not working
when using response.json.
That lead to repeatWhen for example not correctly unsubscribing,
basically fetching forever even after a component was unmounted.

This is also currently happening on pages for repos that are having a cloning status, after leaving the page, they will be fetched until it's done cloning.

(see network tab)
Before:

After:

### CODE REVIEW

**Comment by felixfbecker** [comment]
> That lead to repeatWhen for example not correctly unsubscribing,
basically fetching forever even after a component was unmounted.

This part I still don't understand. If something is unsubscribed, after that point, there can't be more emissions. Whether the underlying request is cancelled shouldn't matter 🤔

**Comment by felixfbecker** [comment]
I am a bit worried that there are subtle differences between XHR and fetch that will introduce unnoticed bugs due to this change or because we start to rely on XHR behavior unconsciously until it's fixed in rxjs (I remember there were things like that when I migrated us).

Could you test if https://github.com/ReactiveX/rxjs/pull/5306 fixes the issue for us? If yes, we should push for that to get merged to minimize the risk.

An alternative would also be to copy the `fromFetch()` implementation in the meantime.

**Comment by eseliger** [comment]
> This part I still don't understand. If something is unsubscribed, after that point, there can't be more emissions. Whether the underlying request is cancelled shouldn't matter 🤔

The observable is never completed properly on outer unsubscription,

```ts
const s = fromFetch(url).pipe(
delay(1000),
finalize(() => console.log('done 1')),
repeatWhen(obs => obs.pipe(delay(5000)),
finalize(() => console.log('done 2'))
).subscribe()
s.unsubscribe()
```

logs

```
done 2
..sleeps..
done 1
```

at least that's what I could figure out while debugging..

File: .eslintrc.js [hunk 1]
```diff
@@ -33,7 +33,21 @@ const config = {
...
-    'no-restricted-imports': ['error', { paths: ['highlight.js', 'marked', 'rxjs/ajax'] }],
+    'no-restricted-imports': [
+      'error',
+      {
+        paths: [
+          'highlight.js',
+          'marked',
+          'rxjs/ajax',
+          {
+            name: 'rxjs/fetch',
```
Comment (felixfbecker) [inline .eslintrc.js]: Good thought thinking of this!

**Pull request merged**